db/pool.go

```diff
@@ -0,0 +1,200 @@
+package db
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"database/sql"
+	"io/ioutil"
+	"net"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/jackc/pgx"
+	"github.com/jackc/pgx/stdlib"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+	"github.com/sean-/vpc/internal/buildtime"
+	"github.com/sean-/vpc/internal/logger"
+)
+
+type Pool struct {
+	pool            *pgx.ConnPool
+	stdDriverConfig *stdlib.DriverConfig
+	config          Config
+}
+
+func New(cfg Config) (*Pool, error) {
+	pool := &Pool{
+		config: cfg,
+	}
+
+	tlsConfig, err := cfg.TLSConfig()
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to generate a TLS config")
+	}
+
+	// TODO(jen20) do we even want to support password auth vs TLS?
```

Comment:
I think it's possible we need to for some cases, like the initial load, but... it should be the exception.  Or passwords should come with a `log.Warn()` periodically in order to shame the operator into using TLS.